extensions/resteasy-reactive/rest/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/QuarkusInvokerFactory.java

```diff
@@ -46,29 +51,36 @@ public Supplier<EndpointInvoker> create(ResourceMethod method, ClassInfo current
 
         String baseName = currentClassInfo.name() + "$quarkusrestinvoker$" + method.getName() + "_"
                 + HashUtil.sha1(endpointIdentifier);
-        try (ClassCreator classCreator = new ClassCreator(
-                new GeneratedClassGizmoAdaptor(generatedClassBuildItemBuildProducer,
-                        applicationClassPredicate.test(currentClassInfo.name().toString())),
-                baseName, null,
-                Object.class.getName(), EndpointInvoker.class.getName())) {
-            MethodCreator mc = classCreator.getMethodCreator("invoke", Object.class, Object.class, Object[].class);
-            ResultHandle[] args = new ResultHandle[method.getParameters().length];
-            ResultHandle array = mc.getMethodParam(1);
-            for (int i = 0; i < method.getParameters().length; ++i) {
-                args[i] = mc.readArrayValue(array, i);
-            }
-            ResultHandle res;
-            if (Modifier.isInterface(currentClassInfo.flags())) {
-                res = mc.invokeInterfaceMethod(info, mc.getMethodParam(0), args);
-            } else {
-                res = mc.invokeVirtualMethod(info, mc.getMethodParam(0), args);
-            }
-            if (info.returnType().kind() == Type.Kind.VOID) {
-                mc.returnValue(mc.loadNull());
-            } else {
-                mc.returnValue(res);
-            }
-        }
+        ClassOutput classOutput = new GeneratedClassGizmo2Adaptor(generatedClassBuildItemBuildProducer, null,
+                applicationClassPredicate.test(currentClassInfo.name().toString()));
+        Gizmo g = Gizmo.create(classOutput);
+        g.class_(baseName, cc -> {
+            cc.defaultConstructor();
+            cc.implements_(EndpointInvoker.class);
+            cc.method("invoke", mc -> {
+                ParamVar resourceParam = mc.parameter("resource", Object.class);
+                ParamVar resourceMethodArgsParam = mc.parameter("args", Object[].class);
+                mc.returning(Object.class);
+                mc.body(bc -> {
+                    List<Expr> args = new ArrayList<>(method.getParameters().length);
+                    Expr res;
+                    for (int i = 0; i < method.getParameters().length; ++i) {
+                        args.add(bc.localVar("var" + i, resourceMethodArgsParam.elem(i)));
```

Comment:
The `localVar` part looks a bit odd to me.
I suppose you didn't add it just for fun but I'm not sure it's very natural. /cc @Ladicek

Reply:
I know I added to get over some issue, but let me check and see if other changes I made have since made it redundant.

Reply:
Seems to be unnecessary, will remove

Reply:
In this particular case, creating locals is not needed, because they are used exactly once and in the correct order, but there's nothing _wrong_ about it either.

Reply:
@Ladicek yeah sure, it's just that since they weren't used outside of this particular call, it seemed a bit useless. So I was wondering if something was off either here or in Gizmo 2.